CMakeLists.txt

```diff
@@ -41,6 +41,8 @@ option(gw_profiling "Compile a binary for profiling with NVTX markers." OFF)
 option(gw_enable_caching_allocator "Enable caching allocator." ON)
 option(gw_generate_docs "Generate Doxygen documentation" ON)
 option(gw_cuda_gen_all_arch "ON: Generate optimized CUDA code for all architectures | OFF: for detected architectures only" OFF)
+# Optionally build htslib for SAM/BAM support. Requires autoconf to be installed
+option(gw_build_htslib "Build 3rdparty htslib that allows output in SAM/BAM format" OFF)
```

Comment:
(See others changes in .cmake files and README.md, as well as the ifdef guards in utils files, main.cu, and sample_cudamapper.cpp that relate to this) @tijyojwad I wasn't sure if this was the _best_ idea, but I made the htslib components optional since they require extra packages to be installed (autoconf). We talked about this before, but I'm not sure how applicable it still is. In any case, I added it to the build & install documentation

Reply:
Yeah I like this optional support capability! Let's keep it